x11-packages/wlroots/build.sh

```diff
@@ -3,22 +3,24 @@ TERMUX_PKG_DESCRIPTION="Modular wayland compositor library"
 TERMUX_PKG_LICENSE="MIT"
 TERMUX_PKG_MAINTAINER="@termux"
 TERMUX_PKG_VERSION="0.16.2"
+TERMUX_PKG_REVISION=1
 TERMUX_PKG_SRCURL=https://gitlab.freedesktop.org/wlroots/wlroots/-/archive/${TERMUX_PKG_VERSION}/wlroots-${TERMUX_PKG_VERSION}.tar.bz2
 TERMUX_PKG_SHA256=afea2cc740344c4e86749cf4908e07692e183cb14a3db854d24dec454c664b88
-TERMUX_PKG_DEPENDS="libdrm, libglvnd, libpixman, libwayland, libxcb, libxkbcommon, xcb-util-renderutil, xcb-util-wm"
+TERMUX_PKG_DEPENDS="libdrm, libglvnd, libpixman, libwayland, libxcb, libxkbcommon, xcb-util-renderutil, xcb-util-wm, vulkan-loader-generic, termux-gui-c"
 TERMUX_PKG_BUILD_DEPENDS="libglvnd-dev, libwayland-cross-scanner, libwayland-protocols, xwayland"
 TERMUX_PKG_RECOMMENDS="xwayland"
+TERMUX_PKG_API_LEVEL=28
```

Comment:
Is there a specific reason why we need to manually set API level 28 here?

Reply:
Used AHardwareBuffer, maybe API26 will also work

Reply:
Yes, [AGardwareBuffer functions are in API 26+](https://developer.android.com/ndk/reference/group/a-hardware-buffer#ahardwarebuffer_acquire)

Reply:
This may not be a clever idea.

It would be better if these functions could be dynamically loaded with `dlopen`, and then it is possible to not override the `TERMUX_PKG_API_LEVEL`.

Reply:
That's what I'm already doing in the `termux-gui-c` library, you can orient yourself on that if you want. You just have to add your private functions, though you'd have to check if they are also in libandroid.so or are in some other VNDK library, I don't know about those.

Sidenode: I heard people say "Supporting multiple API versions on NDK is a pain!". Have they just not heard about `dlopen`? That's also the approach I'm using in the plugin itself to support the feature if available.

Reply:
Okay, now I’ve changed it to dynamically load libandroid.so